lib/contracts/contributionreward.ts

```diff
@@ -157,6 +157,213 @@ export class ContributionRewardWrapper extends ExtendTruffleContract {
     return new ArcTransactionResult(tx);
   }
 
+  /**
+   * Redeem external token reward for the beneficiary of the proposal
+   * @param {ContributionRewardSpecifiedRedemptionParams} opts
+   */
+  public async redeemExternalToken(opts = {}): Promise<ArcTransactionResult> {
+
+    const defaults = {
+      avatar: undefined,
+      proposalId: undefined,
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    if (!options.avatar) {
+      throw new Error("avatar address is not defined");
+    }
+
+    const tx = await this.contract.redeemExternalToken(
+      options.proposalId,
+      options.avatar,
+    );
+
+    return new ArcTransactionResult(tx);
+  }
+
+  /**
+   * Redeem reputation reward for the beneficiary of the proposal
```

Comment:
oh interesting, so anyone can do the redemption and decide to pay the gas for it?

Reply:
Apparently.  I remarked (to myself) about that too.  Same for the other reward types.